ktoml-core/src/commonMain/kotlin/com/akuleshov7/ktoml/tree/nodes/tables/TomlInlineTable.kt

```diff
@@ -146,11 +162,90 @@ public class TomlInlineTable internal constructor(
                     }
                 }
                 .splitInlineTableToKeyValue(config.allowEscapedQuotesInLiteralStrings, lineNo)
-                .map { it.parseTomlKeyValue(lineNo, comments = emptyList(), inlineComment = "", config) }
+                .map {
+                    it.parseTomlKeyValue(lineNo, comments = emptyList(), inlineComment = "", config)
+                }
 
             return parsedList
         }
 
+        private fun String.parseInlineArrayOfTables(
+            keyValuePair: Pair<String, String>,
+            lineNo: Int,
+            config: TomlInputConfig
+        ): List<TomlNode> {
+            val inlineTableValues = this.splitInlineArrayOfTables(config.allowEscapedQuotesInLiteralStrings)
+
+            return inlineTableValues.map { tableValue ->
+                val inlineTableValues = tableValue.parseInlineTableValue(
+                    keyValuePair.first to tableValue.trim(),
+                    lineNo,
+                    config,
+                )
+
+                TomlArrayOfTablesElement(lineNo, emptyList(), "").also { arrayOfTableElement ->
+                    inlineTableValues.forEach { value ->
+                        arrayOfTableElement.appendChild(value)
+                    }
+                }
+            }
+        }
+
+        /**
+         *  Split by "}," - but skip all whitespaces between '}' and ','
+         *  Also ignore characters inside strings and keep closing '}' in each value
+         */
+        @Suppress("NESTED_BLOCK", "TOO_LONG_FUNCTION")
+        private fun String.splitInlineArrayOfTables(allowEscapedQuotesInLiteralStrings: Boolean): List<String> {
+            val clearedString = this
+                .replaceEscaped(allowEscapedQuotesInLiteralStrings)
+                .removePrefix("[")
+                .removeSuffix("]")
+
+            val result: MutableList<String> = mutableListOf()
+            val current = StringBuilder()
+            val lastIdx = clearedString.length
+            var i = 0
+
+            while (i < lastIdx) {
```

Comment:
sorry, for being a nerd, but can you rewrite the code in kotlin style, please 🙏🏾 
can you please use `when` here?

``` 
when (val char = content[i]) {
                '\"', '\'' -> {
```